src/polygonarea.rs

```diff
@@ -1 +1,561 @@
+    }
+
+    #[test]
+    fn test_planimeter12() {
+        // Copied from https://github.com/geographiclib/geographiclib-octave/blob/main/inst/geographiclib_test.m#L701
+        let geoid = Geodesic::wgs84();
+
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+        pa.add_point(66.562222222, 0.0);
+        pa.add_point(66.562222222, 180.0);
+        pa.add_point(66.562222222, 360.0);
+        let (perimeter, area) = pa.compute();
+        assert_relative_eq!(perimeter, 10465729.0, epsilon = 1.0);
+        assert_relative_eq!(area, 0.0);
+    }
+
+    #[test]
+    fn test_planimeter12r() {
+        // Copied from https://github.com/geographiclib/geographiclib-octave/blob/main/inst/geographiclib_test.m#L710
+        let geoid = Geodesic::wgs84();
+
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+
+        pa.add_point(66.562222222, -0.0);
+        pa.add_point(66.562222222, -180.0);
+        pa.add_point(66.562222222, -360.0);
+
+        let (perimeter, area) = pa.compute();
+        assert_relative_eq!(perimeter, 10465729.0, epsilon = 1.0);
+        assert_relative_eq!(area, 0.0);
+    }
+
+    #[test]
+    fn test_planimeter13() {
+        // Copied from https://github.com/geographiclib/geographiclib-octave/blob/main/inst/geographiclib_test.m#L719
+
+        let geoid = Geodesic::wgs84();
+
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+        pa.add_point(89.0, -360.0);
+        pa.add_point(89.0, -240.0);
+        pa.add_point(89.0, -120.0);
+        pa.add_point(89.0, 0.0);
+        pa.add_point(89.0, 120.0);
+        pa.add_point(89.0, 240.0);
+        let (perimeter, area) = pa.compute();
+        assert_relative_eq!(perimeter, 1160741.0, epsilon = 1.0);
+        assert_relative_eq!(area, 32415230256.0, epsilon = 1.0);
+    }
+
+    #[test]
+    fn test_planimeter15() {
+        // Copied from https://github.com/geographiclib/geographiclib-octave/blob/main/inst/geographiclib_test.m#LL728C14-L728C26
+
+        let geoid = Geodesic::wgs84();
+
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+        pa.add_point(2.0, 1.0);
+        pa.add_point(1.0, 2.0);
+        pa.add_point(3.0, 3.0);
+        let (_, area) = pa.compute();
+        assert_relative_eq!(area, 18454562325.45119);
+
+        // Switching the winding
+        let mut pa = PolygonArea::new(&geoid, Winding::Clockwise);
+        pa.add_point(2.0, 1.0);
+        pa.add_point(1.0, 2.0);
+        pa.add_point(3.0, 3.0);
+        let (_, area) = pa.compute();
+        assert_relative_eq!(area, -18454562325.45119);
+
+        // Swaping lat and lon
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+        pa.add_point(1.0, 2.0);
+        pa.add_point(2.0, 1.0);
+        pa.add_point(3.0, 3.0);
+        let (_, area) = pa.compute();
+        assert_relative_eq!(area, -18454562325.45119);
+    }
+
+    #[test]
+    fn test_planimeter21() {
```

Comment:
This looks to be planimeter19, not 21.
```octave
function n = Planimeter19
% Coverage tests, includes Planimeter19 - Planimeter20 (degenerate
% polygons).
  n = 0;
  points = [1,1];
  [area, perimeter] = geodarea(points(:,1), points(:,2));
  n = n + assertEquals(area, 0, 0);
  n = n + assertEquals(perimeter, 0, 0);
end
```

```suggestion
    fn test_planimeter19() {
```

Can you add the real planimeter21?

Reply:
I've added `test_planimeter21`, minus assrtions for `test_poin()` and `test_edge()` which we don't yet support.

Reply:
I've added more degeneracy tests to `test_planimeter19`. 

One thing I'm running into is what perimeter we should be getting for degenerate polygons that are "perfectly flat". 

RIght now if you describe the following polygon:

1. Start at `0.0, 0.0`. 
2. Go due-north to `1.0, 0.0`
3. That's it, that's the polygon

We get the following result:

1. Area is zero: correct
2. Perimeter is `221149 meters` (2 degrees, 1 degree up and one degree down to complete the polygon). This seems like a reasonable perimeter to report for a degenerate polygon.  However, it looks like `geographic-lib` reports perimeter to be `NaN` when the area is zero. 

Thoughts?